staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -760,14 +774,45 @@ type costInfo struct {
 	// that the parent schemas offer no bound to the number of times a data element for the current
 	// schema can exist.
 	MaxCardinality *uint64
+	CRDCost        *crdCost
```

Comment:
document the meaning of this field... IIUC, this is a pointer in order to propagate and modify a shared value as we descend, right?

Reply:
I've added the guards to be super safe

Reply:
I've fleshed out the documentation as well, with notes about how the field lifecycle.